cloudify_hostpool/storage/sqlite.py

```diff
@@ -0,0 +1,123 @@
+            if col[0] == 'auth':
+                # "auth" column value is dictionary serialized to string
+                # -> json.dumps(auth).
+                d[col[0]] = json.loads(row[idx])
+            else:
+                d[col[0]] = row[idx]
+        return d
+
+    def _get_sql_and_values_from_kwargs(self, **kwargs):
+        """ Helper method to create sql query """
+        values = tuple(kwargs.itervalues())
+        sql_part = ' AND '.join('{0}=?'.format(k) for k in kwargs)
+        return sql_part, values
+
+    def get_servers(self, **kwargs):
+        with sqlite3.connect(self._filename) as conn:
+            conn.row_factory = self._dict_factory
+            cursor = conn.cursor()
+            if not kwargs:
+                cursor.execute('SELECT * FROM servers')
+            else:
+                sql_cond, values = self._get_sql_and_values_from_kwargs(
+                    **kwargs)
+                cursor.execute('SELECT * FROM servers WHERE ' + sql_cond,
+                               values)
+            result = cursor.fetchall()
+            return list(result)
+
+    def add_server(self, server):
+        with sqlite3.connect(self._filename) as conn:
+            cursor = conn.cursor()
+            values = (server.get('server_id'), server['address'],
+                      json.dumps(server['auth']), server['port'],
+                      server['alive'], server['reserved'])
+            cursor.execute('INSERT INTO servers '
+                           '(server_id, address, auth, port, alive, reserved) '
+                           'VALUES(?, ?, ?, ?, ?, ?)', values)
+            server['global_id'] = cursor.lastrowid
+            return server
+
+    def update_server(self, server, **kwargs):
+        if not kwargs:
+            return False, None
+        with sqlite3.connect(self._filename) as conn:
+            conn.row_factory = self._dict_factory
+            cursor = conn.cursor()
+            values = tuple(kwargs.itervalues())
+            sql_part = ", ".join('{0}=?'.format(k) for k in kwargs)
+            cursor.execute('UPDATE servers SET {0} WHERE global_id=?'
+                           .format(sql_part),
+                           values + (server['global_id'],))
+            if cursor.rowcount == 0:
+                return False, None
+            conn.commit()
+            cursor.execute('SELECT * FROM servers WHERE global_id=?',
+                           (server['global_id'],))
+            return True, cursor.fetchone()
+
+    def get_server(self, **kwargs):
+        if not kwargs:
+            return None
+        with sqlite3.connect(self._filename) as conn:
+            conn.row_factory = self._dict_factory
+            cursor = conn.cursor()
+            sql_part, values = self._get_sql_and_values_from_kwargs(**kwargs)
+            cursor.execute('SELECT * FROM servers WHERE ' + sql_part, values)
+            return cursor.fetchone()
+
+    def reserve_server(self, server):
```

Comment:
Fixed but corrected the interface. Reserving should work for a server with the id given.